### PR TITLE
Add a "make lint" and fix golint/go vet/gofmt failures

### LINKED ISSUE #176: Fix golint errors in tchannel package
```
> golint . | grep -v _string.go
frame.go:78:1: exported method FrameHeader.MarshalJSON should have comment or be unexported
logger.go:76:1: exported function ErrField should have comment or be unexported
peer_heap.go:85:31: exported method PopPeer returns unexported type *tchannel.peerScore, which can be annoying to use
```

We may want to add a list of packages we check (or check all of them, and have ignore rules e.g. for _string.go files)

### PR DESCRIPTION
**Pull request by prashantv**
Will make this part of Travis in a separate PR to ensure we don't have any regressions.

Fixes #176

### CODE REVIEW

File: pprof/pprof.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ func Register(registrar tchannel.Registrar) {
 		req, err := thttp.ReadRequest(call)
 		if err != nil {
 			registrar.Logger().WithFields(
-				tchannel.LogField{"err", err.Error()},
+				tchannel.LogField{Key: "err", Value: err.Error()},
```
Comment (akshayjshah) [inline pprof/pprof.go]: I can't really argue with the linters, but this makes the API pretty awful.

**Comment by akshayjshah** [comment]
:shipit:

**Comment by jcorbin** [comment]
:+1:

File: pprof/pprof.go (hunk 1 above)
Comment (prashantv) [inline pprof/pprof.go]: +1, might need to change the API to just do `.With` where it takes key/value strings or something.

**Pull request merged**